parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -560,61 +548,57 @@ loop = do
         -- `mdValues` is (names of the) metadata to pass to `op`
         -- `op` is the operation to add/remove/alter metadata mappings.
         --   e.g. `Metadata.insert` is passed to add metadata links.
-        manageLinks :: Bool
-                    -> [(Path', HQ'.HQSegment)]
-                    -> [HQ.HashQualified Name]
-                    -> (forall r. Ord r
-                        => (r, Metadata.Type, Metadata.Value)
-                        ->  Branch.Star r NameSegment
-                        ->  Branch.Star r NameSegment)
-                    -> Action m (Either Event Input) v ()
+        manageLinks ::
+          Bool ->
+          [(Path', HQ'.HQSegment)] ->
+          [HQ.HashQualified Name] ->
+          ( forall r.
+            Ord r =>
+            (r, Metadata.Type, Metadata.Value) ->
+            Branch.Star r NameSegment ->
+            Branch.Star r NameSegment
+          ) ->
+          Action m (Either Event Input) v ()
         manageLinks silent srcs mdValues op = do
-          mdValuels <- fmap (first toList) <$>
-            traverse (\x -> fmap (,x) (getHQTerms x)) mdValues
-          before <- Branch.head <$> use root
-          traverse_ go mdValuels
-          after  <- Branch.head <$> use root
-          (ppe, outputDiff) <- diffHelper before after
-          if not silent then
-            if OBranchDiff.isEmpty outputDiff
-            then respond NoOp
-            else respondNumbered $ ShowDiffNamespace Path.absoluteEmpty
-                                                     Path.absoluteEmpty
-                                                     ppe
-                                                     outputDiff
-          else unless (OBranchDiff.isEmpty outputDiff) $
-                 respond DefaultMetadataNotification
+          runExceptT (for mdValues \val -> ExceptT (getMetadataFromName val)) >>= \case
+            Left output -> respond output
+            Right metadata -> do
+              before <- Branch.head <$> use root
+              traverse_ go metadata
+              if silent
+                then respond DefaultMetadataNotification
+                else do
+                  after <- Branch.head <$> use root
```

Comment:
No change in behavior in the "else" branch here. The key difference is that we only perform a branch diff here, but not in the "then" branch.